### PR TITLE
Add ref finance example

### PR DESCRIPTION
**Pull request by bgodlin**

### CODE REVIEW

**Review by deverka — APPROVED** [review]

**Comment by deverka** [comment]
@jamesbayly please take a look

File: Near/near-ref-finance/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# SubQuery - Example Project for NEAR
```
Comment (jamesbayly) [inline Near/near-ref-finance/README.md]:
```suggestion
# SubQuery - Example Project for Ref Finance on NEAR
```

File: Near/near-ref-finance/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      DB_HOST: postgres
+      DB_PORT: 5432
+    volumes:
+      - ./:/app
+    command:
+      - ${SUB_COMMAND:-} # set SUB_COMMAND env variable to "test" to run tests
+      - -f=/app
+      - --db-schema=app
+      - --workers=4
+      - --batch-size=1
```
Comment (jamesbayly) [inline Near/near-ref-finance/docker-compose.yml]:
```suggestion
```

Don't commit this kind of `batch-size`

File: Near/near-ref-finance/project.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    },
+  },
+  description:
+    "This is an example project that indexes price oracle feeds from the NEAR blockchain using SubQuery",
+  repository: "https://github.com/subquery/near-subql-starter",
+  schema: {
+    file: "./schema.graphql",
+  },
+  network: {
+    // chainId is the EVM Chain ID, for Near Aurora this is 1313161554
```
Comment (jamesbayly) [inline Near/near-ref-finance/project.ts]: This looks incorrect?

File: Near/near-ref-finance/src/mappings/mappingHandlers.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import { Swap, Token, Pool } from "../types";
+import { NearAction } from "@subql/types-near";
+
+export async function handleAction(action: NearAction): Promise<void> {
+  // An Action can belong to either a transaction or a receipt
```
Comment (jamesbayly) [inline Near/near-ref-finance/src/mappings/mappingHandlers.ts:6]: I think it's good practice to keep a logger event here to clearly show to people that "this is working"
```suggestion
  logger.info(`Swap found at block ${action.transaction.block_height}`);
  // An Action can belong to either a transaction or a receipt
```

File: Near/near-ref-finance/src/mappings/mappingHandlers.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      id: `${action.transaction.block_height}-${action.transaction.result.id}-${action.id}-${i}`,
+      poolId: (await getOrCreatePool(JSON.stringify(actions[i]["pool_id"]))).id,
+      firstTokenId: (
+        await getOrCreatePool(JSON.stringify(actions[i]["token_in"]))
+      ).id,
+      secondTokenId: (
+        await getOrCreatePool(JSON.stringify(actions[i]["token_out"]))
+      ).id,
+    }).save();
+    logger.warn("Swap is saved");
```
Comment (jamesbayly) [inline Near/near-ref-finance/src/mappings/mappingHandlers.ts]:
```suggestion
    logger.info("Swap is saved");
```

File: Near/near-ref-finance/src/mappings/mappingHandlers.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  }
+}
+
+export async function getOrCreateToken(tokenid: string): Promise<Token> {
```
Comment (jamesbayly) [inline Near/near-ref-finance/src/mappings/mappingHandlers.ts]: do these need to be exported?

File: Near/near-ref-finance/src/mappings/mappingHandlers.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  let token = await Token.get(tokenid);
+  if (token === undefined) {
+    token = Token.create({ id: tokenid });
+    await token.save();
+  }
+  return token;
+}
+
+export async function getOrCreatePool(poolid: string): Promise<Pool> {
```
Comment (jamesbayly) [inline Near/near-ref-finance/src/mappings/mappingHandlers.ts]: Same here?

**Review by jamesbayly — CHANGES_REQUESTED** [review]

File: Near/near-ref-finance/project.ts (hunk 3 above)
Comment (bgodlin) [reply Near/near-ref-finance/project.ts]: Resolved in https://github.com/subquery/near-subql-starter/pull/9/commits/17daedf84833222b353afcf060f7fdbe8381d373

File: Near/near-ref-finance/src/mappings/mappingHandlers.ts (hunk 6 above)
Comment (bgodlin) [reply Near/near-ref-finance/src/mappings/mappingHandlers.ts]: Resolved as part of https://github.com/subquery/near-subql-starter/pull/9/commits/8432ad594fbf8bb2155396748cd1179fd74f59e5

File: Near/near-ref-finance/src/mappings/mappingHandlers.ts (hunk 7 above)
Comment (bgodlin) [reply Near/near-ref-finance/src/mappings/mappingHandlers.ts]: Resolved as part of https://github.com/subquery/near-subql-starter/commit/8432ad594fbf8bb2155396748cd1179fd74f59e5

**Pull request merged**